<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>744</width>
    <height>614</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MLP</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QCheckBox" name="check_box_cross_validation">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>510</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Cross validation</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>240</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="Hidden_layers_layout">
     <item>
      <widget class="QRadioButton" name="radio_button_5">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>5</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radio_button_4">
       <property name="text">
        <string>4</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radio_button_3">
       <property name="text">
        <string>3</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radio_button_2">
       <property name="text">
        <string>2</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="hidden_number_label">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>210</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of hidden layers</string>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>180</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="Perceptron_type_layout">
     <item>
      <widget class="QRadioButton" name="radio_button_graph_type">
       <property name="text">
        <string>Matrix</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radio_button_matrix_type">
       <property name="text">
        <string>Graph</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="load_weights_button">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>270</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="text">
     <string>Load weight</string>
    </property>
   </widget>
   <widget class="Line" name="line_vertical">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>-20</y>
      <width>20</width>
      <height>611</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="lineWidth">
     <number>3</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_horizontal_1">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>310</y>
      <width>211</width>
      <height>20</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="lineWidth">
     <number>3</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_learn">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>430</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>25</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Learn</string>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_settings">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>130</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>25</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Settings</string>
    </property>
   </widget>
   <widget class="QPushButton" name="start_learning">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>550</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Start learning</string>
    </property>
   </widget>
   <widget class="QPushButton" name="save_weights_button">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>270</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Save weight</string>
    </property>
   </widget>
   <widget class="Line" name="line_horizontal_2">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>420</y>
      <width>201</width>
      <height>20</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="lineWidth">
     <number>3</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="start_testing">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>380</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Start testing</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_test">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>330</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>25</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Test</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="sample_part">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>360</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="decimals">
     <number>1</number>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_sample_part">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>360</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Sample part</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="crossv_groups_count">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>490</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>5</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_number_of_groups">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>490</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of groups</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_number_of_epochs">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>470</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of epochs</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="epochs_count">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>470</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>5</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
   </widget>
   <widget class="QCheckBox" name="check_box_learn_report">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>530</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Report</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="s21::PaintWidget" name="draw_widget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>512</width>
      <height>512</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <widget class="QStackedWidget" name="stackedWidget_report">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>10</y>
       <width>461</width>
       <height>471</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="lineWidth">
      <number>2</number>
     </property>
     <property name="currentIndex">
      <number>4</number>
     </property>
     <widget class="QWidget" name="test_report">
      <widget class="QLabel" name="label_report">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>5</y>
         <width>131</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>25</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Test report</string>
       </property>
       <property name="buddy">
        <cstring>label_report</cstring>
       </property>
      </widget>
      <widget class="QLabel" name="label_accuracy">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>351</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Average accuracy</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_precision">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>351</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Precision</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_recal">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>381</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Recall</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_f_measure">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>110</y>
         <width>391</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>f-measure</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_time">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>130</y>
         <width>391</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Time</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="learning_report">
      <widget class="QLabel" name="label_report_learing">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>5</y>
         <width>181</width>
         <height>41</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>25</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Learning report</string>
       </property>
       <property name="buddy">
        <cstring>label_report</cstring>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="answer">
      <widget class="QLabel" name="label_answer">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>-10</y>
         <width>71</width>
         <height>81</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>60</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="loading_window">
      <widget class="QLabel" name="label_loading">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>170</y>
         <width>211</width>
         <height>91</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>45</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Complited</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="meassage">
      <widget class="QLabel" name="label_message">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>130</y>
         <width>441</width>
         <height>181</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>30</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>File weight error</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="Line" name="line_horizontal_3">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>110</y>
      <width>201</width>
      <height>20</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="lineWidth">
     <number>3</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_recognise">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>0</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_clear">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_load_img">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>0</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_image">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>0</y>
      <width>181</width>
      <height>111</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_close_report">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>0</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="type_label">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>160</y>
      <width>96</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Perceptron type</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="check_box_test_report">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>390</y>
      <width>85</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Report</string>
    </property>
   </widget>
   <zorder>check_box_cross_validation</zorder>
   <zorder>horizontalLayoutWidget</zorder>
   <zorder>hidden_number_label</zorder>
   <zorder>horizontalLayoutWidget_2</zorder>
   <zorder>line_horizontal_1</zorder>
   <zorder>label_learn</zorder>
   <zorder>start_learning</zorder>
   <zorder>save_weights_button</zorder>
   <zorder>line_horizontal_2</zorder>
   <zorder>start_testing</zorder>
   <zorder>label_test</zorder>
   <zorder>sample_part</zorder>
   <zorder>label_sample_part</zorder>
   <zorder>crossv_groups_count</zorder>
   <zorder>label_number_of_groups</zorder>
   <zorder>label_number_of_epochs</zorder>
   <zorder>epochs_count</zorder>
   <zorder>check_box_learn_report</zorder>
   <zorder>draw_widget</zorder>
   <zorder>line_horizontal_3</zorder>
   <zorder>pushButton_recognise</zorder>
   <zorder>pushButton_clear</zorder>
   <zorder>pushButton_load_img</zorder>
   <zorder>label_image</zorder>
   <zorder>pushButton_close_report</zorder>
   <zorder>label_settings</zorder>
   <zorder>line_vertical</zorder>
   <zorder>type_label</zorder>
   <zorder>load_weights_button</zorder>
   <zorder>check_box_test_report</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>744</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::PaintWidget</class>
   <extends>QWidget</extends>
   <header>view/paint_widget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
